crates/tools/src/prepare_scripts.rs

```diff
@@ -0,0 +1,446 @@
+        tron_account_lock: get_path("tron_account_lock"),
+        meta_contract_validator: get_path("meta_contract_validator"),
+        polyjuice_validator: get_path("polyjuice_validator"),
+        state_validator_lock: get_path("state_validator_lock"),
+        poa_state: get_path("poa_state"),
+    };
+    let build_scripts_result = BuildScriptsResult {
+        programs,
+        lock: Script {
+            code_hash: H256::default(),
+            hash_type: ScriptHashType::Data,
+            args: JsonBytes::default(),
+        },
+        built_scripts: scripts_info
+            .iter()
+            .map(|(k, v)| (k.clone(), v.target_script_path(target_dir)))
+            .collect(),
+    };
+    let output_content =
+        serde_json::to_string_pretty(&build_scripts_result).expect("serde json to string pretty");
+    let output_dir = output_path.parent().expect("get output dir");
+    fs::create_dir_all(&output_dir).expect("create output dir");
+    fs::write(output_path, output_content.as_bytes())?;
+    log::info!("Finish");
+    Ok(())
+}
+
+fn build_godwoken_scripts(repos_dir: &Path, repo_name: &str) {
+    let repo_dir = make_path(repos_dir, vec![repo_name]).display().to_string();
+    let target_dir = format!("{}/c", repo_dir);
+    run_command("make", vec!["-C", &target_dir]).expect("run make");
+    run_command_in_dir(
+        "capsule",
+        vec!["build", "--release", "--debug-output"],
+        &repo_dir,
+    )
+    .expect("run capsule build");
+}
+
+fn build_godwoken_polyjuice(repos_dir: &Path, repo_name: &str) {
+    let target_dir = make_path(repos_dir, vec![repo_name]).display().to_string();
+    run_command("make", vec!["-C", &target_dir, "all-via-docker"]).expect("run make");
+}
+
+fn build_clerkb(repos_dir: &Path, repo_name: &str) {
+    let target_dir = make_path(repos_dir, vec![repo_name]).display().to_string();
+    run_command("yarn", vec!["--cwd", &target_dir]).expect("run yarn");
+    run_command("make", vec!["-C", &target_dir, "all-via-docker"]).expect("run make");
+}
+
+fn collect_scripts_to_target(
+    repos_dir: &Path,
+    target_dir: &Path,
+    scripts_info: &HashMap<String, ScriptsInfo>,
+) {
+    scripts_info.iter().for_each(|(_, v)| {
+        let target_path = v.target_script_path(target_dir);
+        let source_path = v.source_script_path(repos_dir);
+        fs::create_dir_all(&target_path.parent().expect("get dir")).expect("create scripts dir");
+        log::debug!("copy {:?} to {:?}", source_path, target_path);
+        fs::copy(source_path, target_path).expect("copy script");
+    });
+}
+
+fn run_pull_code(mut repo_url: Url, is_recursive: bool, repos_dir: &Path, repo_name: &str) {
+    let commit = repo_url
+        .fragment()
+        .expect("valid branch, tag, or commit")
+        .to_owned();
+    repo_url.set_fragment(None);
+    let target_dir = make_path(repos_dir, vec![repo_name]);
+    if run_git_checkout(&target_dir.display().to_string(), &commit).is_ok() {
+        return;
```

Comment:
Why not raise errors?

```
[2021-06-01T15:22:47Z INFO  gw_tools::prepare_scripts] [Execute]: git ["-C", "tmp/scripts-build-dir/godwoken-polyjuice", "fetch"]
fatal: cannot change to 'tmp/scripts-build-dir/godwoken-polyjuice': No such file or directory
[2021-06-01T15:22:47Z INFO  gw_tools::prepare_scripts] [Execute]: git ["clone", "https://github.com/nervosnetwork/godwoken-polyjuice", "tmp/scripts-build-dir/godwoken-polyjuice", "--recursive"]
```

`git checkout` is failed ,but the prepare-scripts process is not terminated.